pilot/pkg/model/service.go

```diff
@@ -825,25 +825,29 @@ func GetServiceAccounts(svc *Service, ports []int, discovery ServiceDiscovery) [
 }
 
 // DeepCopy creates a clone of Service.
-// TODO : See if there is any efficient alternative to this function - copystructure can not be used as is because
-// Service has sync.RWMutex that can not be copied.
 func (s *Service) DeepCopy() *Service {
```

Comment:
Wonder if for our DeepCopy implementations we should add Fuzz tests comparing them to copystructure or something. Then we can ensure our performance optimizations are also correct

Reply:
SGTM